### PR TITLE
added ssh support and POWERLEVEL9K_HIDE_HOST feature

### PR DESCRIPTION
**Pull request by shibumi**
* refactorized the sourcecode
* added true/false switch

### CODE REVIEW

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -434,13 +434,27 @@ prompt_battery() {
...
+  if [[ "$USER" != "$DEFAULT_USER" ]]; then
     if [[ $(print -P "%#") == '#' ]]; then
-      # Shell runs as root
-      "$1_prompt_segment" "$0_ROOT" "$2" "$DEFAULT_COLOR" "yellow" "$USER@%m"
+      current_state="ROOT"
+    fi
+    if [[ -z "$SSH_CLIENT" && -z "$SSH_TTY" ]]; then
+      if [[ "$POWERLEVEL9K_HIDE_HOST" == "false" || -z "$POWERLEVEL9K_HIDE_HOST"  ]]; then
+        content="${content}@%m"
+      else
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: This else branch is IMHO unnecessary. `content` is already defined.

**Comment by dritter** [comment]
Looks good to me. Nice work @shibumi 👍

**Comment by shibumi** [comment]
I fixed the else branch

**Comment by bhilburn** [comment]
Aha - in the process of testing this, I found an issue:

So, for the most part, I use the same username on most systems I access. The result of this is that my `$DEFAULT_USER` is almost always the same thing - `bhilburn`. The way this change is written now, it will only display the context if you are not using your `$DEFAULT_USER`, even if you're in an SSH session.

I think @shibumi's idea is awesome, and I think we should take it a step further. I think we should make it an option to always show the context segment if you are in an active SSH session.

Thoughts, @shibumi?

**Comment by shibumi** [comment]
@bhilburn 
You mean we should have this SSH feature also for users with `$DEFAULT_USER`?
Mhh good idea. Additional to this, I have also a context trigger in the right-bar as you can see in one of my screenshots.

https://github.com/bhilburn/powerlevel9k/issues/336

**Comment by bhilburn** [comment]
@shibumi - Yeah, that context trigger is slick.

And yes, exactly. Even if the user is `$DEFAULT_USER`, I actually want to indicate an SSH session if one is active. Is this something you can add to this PR before I merge it, or would you like to get this work merged and then add that separately?

**Comment by shibumi** [comment]
@bhilburn I have an additional identifier.. is this a solution for you?

**Comment by bhilburn** [comment]
@shibumi - So that's an additional segment that turns on anytime the terminal exists in an active SSH session?

**Comment by bhilburn** [comment]
@shibumi - I'd love to get this merged! Just a ping on your 'ssh' segment. Also, it looks like branch has conflicts with our current HEAD on `next`.

**Comment by shibumi** [comment]
@bhilburn Yep, thats an additional segment that turn on anytime the terminal exists in an active SSH Session. I have realized this via custom function:
```
zsh_detect_ssh(){
    local color='%F{yellow}'
    if [ -n "$SSH_CLIENT" ] || [ -n "$SSH_TTY" ]; then
        echo -n "%{$color%} ssh"
    fi
}
```

I have no idea how I can include this into the right segment on default.  I would like to have this additional SSH segment in an additional pull request. Or do you think I should have one pull request for both?

Let me checkout the conflicts.. My current branches are a little bit chaotic at the moment.

**Comment by bhilburn** [comment]
@shibumi - Ah, I see, awesome. Can you push your custom segment as a new segment in this PR? As soon as that's up I want to pull this branch and play with it. I think it's going to be a really great new feature for sysadmin users.

**Comment by bhilburn** [comment]
@shibumi - Did you ever get a chance to try and clear up the conflicts & add your custom segment to the branch, by chance? It would be great to get this merged `=)`

**Comment by shibumi** [comment]
@bhilburn - Hey, sorry. I am little busy the last days. Let me finish my math exam at university at wednesday.Then I will be full in again and will hopefully make this PR ready :)

**Comment by dritter** [comment]
Hey @shibumi !
I hope your math exam did go well. Have you already found some time to finish this PR?

**Comment by shibumi** [comment]
@dritter 
Working on it now :) and yes the math exam did go well.

**Comment by shibumi** [comment]
@dritter @bhilburn 
Ok Done. 

But I have one question about the `$1_prompt_segment`
```
"$1_prompt_segment" "$0" "$2" "$DEFAULT_COLOR" "$color" "ssh"
```

`DEFAULT_COLOR` is here the background color right? If so, I got this wrong in the detect_virt prompt..

**Comment by dritter** [comment]
@shibumi Excellent! 
Yep, the third parameter is the background color.. Why is it wrong in the `detect_virt` segment? Just because there `$DEFAULT_COLOR` and `$color` switched positions?
Well that is just like saying "print this segment black on white" or "print this segment white on black" (assuming that `$color` and `$DEFAULT_COLOR` is white and black; in your case it would be rather black and yellow).

**Comment by shibumi** [comment]
@dritter 
It's wrong because black font on yellow background looks ugly (for me). The SSH and the detect_virt segment have both a default color and no way to change this. I think I should add a variable for changing this. At the moment both are supposed to be yellow font on black background.

**Comment by dritter** [comment]
Try setting
```zsh
POWERLEVEL9K_DETECT_VIRT_FOREGROUND='red'
POWERLEVEL9K_DETECT_VIRT_BACKGROUND='green'
```
😉

**Comment by rjorgenson** [comment]
@shibumi I believe all colors can be overwritten with the POWERLEVEL9K_{SEGMENT_NAME}_{FORE|BACK}GROUND variables. SEGMENT_NAME is not always the just the name of the segment, sometimes it incorporates a state, for example POWERLEVEL9K_DIR_HOME_FOREGROUND. There may be edge cases but this works for every state I use and looking at the way you call your segment it should work for this as well =]

**Comment by shibumi** [comment]
@bhilburn I think this one here is ready to merge :) except you have something to complain about :D

**Comment by bhilburn** [comment]
@shibumi - Thanks for the updates, here! After test driving it for a bit, I decided to more or less merge the functionality, and instead made the `context` segment conditionalize on whether or not an SSH session was active at the top-level, and then tried to simplify the code as much as possible.

Give it a shot and let me know if it works for you! Great idea, by the way. I'm really happy you submitted this!

**Pull request merged**